### PR TITLE
Common_test : Updated grammar for tests specifications

### PR DESCRIPTION
**Pull request by acw224**
CommonTest : Update the grammar for test_spec and add relevant tests as documented in #5088

### CODE REVIEW

**Comment by IngelaAndin** [comment]
I think this sounds like a good idea. We are still working to get up to full speed again after summer vacations so, please have some patience. We will start by testing your branch and hopefully have some more feedback fairly soon.

File: lib/common_test/src/ct_run.erl [hunk 1]
```diff
@@ -2050,8 +2050,29 @@ final_tests1([{TestDir,Suite,GrsOrCs}|Tests], Final, Skip, Bad) when
...
+				 				({GroupName,_,_}, Groups)
+							 -> [GroupName | Groups];
+							    (GroupName, Groups)
+							 -> [GroupName | Groups] end,
+													 [], GroupList),
+						 [ct_groups:make_conf(TestDir, Suite,
+				 				[GroupNames], Props, TCs)];
+				 _ -> [ct_groups:make_conf(TestDir, Suite,
+				 GroupOrGroups, [], TCs)]
+			end;
```
Comment (IngelaAndin) [inline lib/common_test/src/ct_run.erl:2075]: This is rather hard to read. For instance if you format the fun like 
```
fun({_,_} = GrSpec, Props) -> 
		[GrSpec | Props];
	   ({_,_,_} = GrSpec, Props) -> 
		[GrSpec | Props];
	   (_, Props) -> 
		Props 
	end,
```     it becomes easier to parse.  Also moving GroupNames first and then refering to it in "Props construction" as well as later, I think would make the code easier to read. Even without my suggest changes the indentation does not follow the normal indentation rules.

**Comment by u3s** [comment]
- please fix the build error
- you might consider branch renaming, to make it similar to OTP convention - e.g. "acw224/ct/group_parsing_modification"

**Comment by acw224** [comment]
Updated with new branch name in #5242

**Pull request closed without merging**